File: shafaet/binary_search.py
def binary_search(iterable, n):
    index = None
    lower = 0
    upper = len(iterable)
    while lower <= upper:
        mid = (lower + upper) // 2
        temp = iterable[mid]
        if temp == n:
            index = mid
            break
        elif temp < n:
            lower = mid - 1
        else:
            upper = mid + 1
    return index



if __name__ == '__main__':
    data = [10, 20, 30, 35, 40, 50, 60]
    n = 30
    print(binary_search(data, n))

File: AlgoExpert/binary-search.go
package main

func BinarySearch(array []int, target int) int {
	i, j := 0, len(array)-1
	for i <= j {
		mid := (i + j) / 2
		value := array[mid]
		if value == target {
			return mid
		} else if value > target {
			j = mid - 1
		} else {
			i = mid + 1
		}
	}
	return -1
}
